### ISSUE TITLE
Bangla font rendering problem with letter ড় ঢ় in LibreOffice 6.0 with some bengali fonts

### ISSUE THREAD

**Issue by kuntalkundu** [harfbuzz/harfbuzz#779]
I reported this issue on https://bugs.documentfoundation.org/show_bug.cgi?id=115421 on 3rd Feb, 2018. (Bug Id- 115421). In reply of this bug report, Mr. Khaled Hosny told me to report this bug here, so I did so. The problem as I told early is problem with Nukta sign. Libre Office use this character to create few bengali alphabet like  ড় ঢ় where the round dots of those letters become slightly displaced with some bengali fonts. But I test those fonts in alternative office packages like OpenOffice, MS Office where the Nukta character doesn't use to form those alphabets. My request is please fix this problem of LibreOffice in next update. I attaced few screenshots and few font which are used

.
![27332268_1789682217708983_7452483768167553570_n](https://user-images.githubusercontent.com/28354301/36074574-de6ca59a-0f67-11e8-8f8d-d0fb53dcfdfb.jpg)
![27336372_1789682214375650_3125569294339190204_n](https://user-images.githubusercontent.com/28354301/36074575-deae0486-0f67-11e8-8072-5bb75a6aee0a.jpg)

**Comment by KrasnayaPloshchad** [comment]
Behdad, you can get the font from [here](https://bugs.documentfoundation.org/attachment.cgi?id=139551) for test.

**Comment by KrasnayaPloshchad** [comment]
CC @jfkthame

**Comment by KrasnayaPloshchad** [comment]
OK, I tested with `hb-shape`, there are different behaviors between HarfBuzz and Uniscribe.
```
harfbuzz-1.7.5-win32\hb-shape.exe "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC"  --shapers=ot
[glyph521=0+1011|space=1+582|glyph522=2+756|space=3+582|glyph521=4+1011|space=6+582|glyph522=7+756]

harfbuzz-1.7.5-win32\hb-shape.exe "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC"  --shapers=uniscribe
[bn_rha=0+976|space=1+582|bn_yya=2+763|space=3+582|glyph521=4+1011|space=6+582|glyph522=7+756]
```
So I think this is probably HarfBuzz bug.

**Comment by KrasnayaPloshchad** [comment]
I made another test for this.
`harfbuzz-1.7.5-win32\hb-view "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC" --shapers=ot --output-format=png --output-file=test1.png`
![test1](https://user-images.githubusercontent.com/10364351/37100143-32b5f266-225d-11e8-9192-9251c0b25fb3.png)
`harfbuzz-1.7.5-win32\hb-view "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC" --shapers=uniscribe --output-format=png --output-file=test2.png`
![test2](https://user-images.githubusercontent.com/10364351/37100145-33371a94-225d-11e8-90e7-f9da8837d199.png)
On Windows, LibreOffice switched to HarfBuzz since 2016 when LO 5.3 is under heavy development, but AOO still relying on Uniscribe, thus your screenshots are in fact showing how does HarfBuzz and Uniscribe behavoired in this case. BTW it’s so weird that dda + nukta ≠ rra in this font.

**Comment by KrasnayaPloshchad** [comment]
CC @khaledhosny

**Comment by kuntalkundu** [comment]
I am really thankfull to team Harfbuzz for this concern. Your team took this font issue seriously. You can test this font in fontforge by reshaping the size, width of nukta dot. Then you can realize the issue. Thanks on advance.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by KrasnayaPloshchad** [comment]
CC @ebraminio

**Comment by ebraminio** [comment]
Confirmed the issue using your test case,

```
$ WINEDLLOVERRIDES="usp10=n" wine winbuild/util/hb-shape.exe "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC"  --shapers=uniscribe

[bn_rha=0+976|space=1+582|bn_yya=2+763|space=3+582|glyph521=4+1011|space=6+582|glyph522=7+756]

$ wine winbuild/util/hb-shape.exe "Bangla Akademi.ttf" --unicodes="09DC 0020 09DD 0020 09A1 09BC 0020 09A2 09BC" --shapers=ot

[glyph521=0+1011|space=1+582|glyph522=2+756|space=3+582|glyph521=4+1011|space=6+582|glyph522=7+756]
```

@KrasnayaPloshchad I like to help on Indic shaping but to be honest I just don't know what is going on :P Anyone interested on these and have free time, feel free to contact me privately, using an instant messenger to talk about these is the best option for me.

**Comment by ebraminio** [comment]
I guess I've found a fix on #977

**Comment by ebraminio** [comment]
But it would be super useful if you can find a free font with the issue.
